sdks/java/io/hdfs/src/main/java/org/apache/beam/sdk/io/hdfs/HDFSFileSink.java

```diff
@@ -237,7 +239,16 @@ public HDFSWriter(HDFSWriteOperation<K, V> writeOperation,
     }
 
     @Override
-    public void open(String uId) throws Exception {
+    public final void open(String uId,
+                           @Nullable BoundedWindow window,
+                           @Nullable PaneInfo paneInfo,
+                           int shard,
+                           int numShards) throws Exception {
+      if (window != null) {
+        throw new UnsupportedOperationException("Windowing support not implemented yet for"
```

Comment:
I think it would be a bit more actionable to indicate what window was passed, here.